File: Small_ProgrammingLang/SourceCode/MachineUI/FormDisplayConsole.vb


Public Class FormDisplayConsole


    Public Sub Print(ByRef Userinput As String)
        Me.Zx81_DisplayScreen.Text &= Userinput
    End Sub
    Public Sub CLS()
        Me.Zx81_DisplayScreen.Text = "" & vbCr
    End Sub
    Public Function Input(ByRef Message As String) As String
        '    Default = "1"    ' Set default.
        ' Display message, title, and default value.
        Return InputBox(Message, "INPUT")
    End Function

End Class

File: Small_ProgrammingLang/C_/SourceCode/MachineUI/FormDisplayConsole.cs
using Microsoft.VisualBasic;

namespace SDK
{
    public partial class FormDisplayConsole
    {
        public FormDisplayConsole()
        {
            InitializeComponent();
        }

        public void Print(ref string Userinput)
        {
            Zx81_DisplayScreen.Text += Userinput;
        }

        public void CLS()
        {
            Zx81_DisplayScreen.Text = "" + Constants.vbCr;
        }

        public string Input(ref string Message)
        {
            // Default = "1"    ' Set default.
            // Display message, title, and default value.
            return Interaction.InputBox(Message, "INPUT");
        }
    }
}